framework/src/base/MooseServer.C

```diff
@@ -0,0 +1,529 @@
+  (void)include_declaration; // suppress unused variable warning
+
+  if (!this->is_setup)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be setup" << std::endl;
+    return false;
+  }
+
+  if (!this->is_initialized)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be initialized" << std::endl;
+    return false;
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  // TODO - hook up this capability by adding server specific logic here
+
+  return pass;
+}
+
+bool
+MooseServer::gatherDocumentFormattingTextEdits(wasp::DataArray & formattingTextEdits,
+                                               int start_line,
+                                               int start_character,
+                                               int end_line,
+                                               int end_character,
+                                               int tab_size,
+                                               bool insert_spaces)
+{
+  (void)formattingTextEdits; // suppress unused variable warning
+  (void)start_line;          // suppress unused variable warning
+  (void)start_character;     // suppress unused variable warning
+  (void)end_line;            // suppress unused variable warning
+  (void)end_character;       // suppress unused variable warning
+  (void)tab_size;            // suppress unused variable warning
+  (void)insert_spaces;       // suppress unused variable warning
+
+  if (!this->is_setup)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be setup" << std::endl;
+    return false;
+  }
+
+  if (!this->is_initialized)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be initialized" << std::endl;
+    return false;
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  // TODO - hook up this capability by adding server specific logic here
+
+  return pass;
+}
+
+bool
+MooseServer::gatherDocumentSymbols(wasp::DataArray & documentSymbols)
+{
+  if (!this->is_setup)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be setup" << std::endl;
+    return false;
+  }
+
+  if (!this->is_initialized)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server needs to be initialized" << std::endl;
+    return false;
+  }
+
+  if (!this->is_document_open)
+  {
+    this->errors << wasp::lsp::m_error_prefix << "Server has no open document" << std::endl;
+    return false;
+  }
+
+  bool pass = true;
+
+  if (!this->check_app || !this->check_app->parser()._root)
+  {
+    return pass;
+  }
+
+  wasp::HITNodeView view_root = this->check_app->parser()._root->getNodeView();
+
+  for (std::size_t i = 0, count = view_root.child_count(); i < count; i++)
```

Comment:
```suggestion
  for (const auto i: make_range(view_root.child_count())
```

is the new way of setting up loop heders

Reply:
Good feedback - I just added a commit addressing this.